build fail looks real. It might have to do with a hack I've used to mock out requests in tests:

```
======================================================================
ERROR: testDefaultValue (casexml.apps.stock.tests.test_consumption_for_case.ConsumptionCaseTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "corehq/ex-submodules/casexml/apps/stock/tests/test_consumption_for_case.py", line 26, in testDefaultValue
    self._stock_report(25, 5)
  File "corehq/ex-submodules/casexml/apps/stock/tests/base.py", line 45, in _stock_report
    txn.save()
  File "corehq/ex-submodules/casexml/apps/stock/models.py", line 71, in save
    force_insert=force_insert, force_update=force_update, using=using, update_fields=update_fields
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/django/db/models/base.py", line 589, in save
    force_update=force_update, update_fields=update_fields)
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/django/db/models/base.py", line 626, in save_base
    update_fields=update_fields, raw=raw, using=using)
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/django/dispatch/dispatcher.py", line 198, in send
    response = receiver(signal=self, sender=sender, **named)
  File "corehq/ex-submodules/casexml/apps/stock/signals.py", line 19, in stock_transaction_post_save_signal_receiver
    update_stock_state_for_transaction(instance)
  File "corehq/ex-submodules/casexml/apps/stock/signals.py", line 77, in update_stock_state_for_transaction
    domain = Domain.get_by_name(domain_name)
  File "/home/travis/build/dimagi/commcare-hq/corehq/util/quickcache.py", line 308, in inner
    return helper(*args, **kwargs)
  File "/home/travis/build/dimagi/commcare-hq/corehq/util/quickcache.py", line 197, in __call__
    return super(SkippableQuickCache, self).__call__(*args, **kwargs)
  File "/home/travis/build/dimagi/commcare-hq/corehq/util/quickcache.py", line 97, in __call__
    content = self.fn(*args, **kwargs)
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/domain/models.py", line 524, in get_by_name
    _assert(False, '%r is not a valid domain name' % name)
  File "/home/travis/build/dimagi/commcare-hq/corehq/util/soft_assert/core.py", line 24, in __call__
    return self._call(assertion, msg, obj)
  File "/home/travis/build/dimagi/commcare-hq/corehq/util/soft_assert/core.py", line 49, in _call
    key=tb_id, line=line, obj=obj))
  File "/home/travis/build/dimagi/commcare-hq/corehq/util/soft_assert/api.py", line 90, in send
    _send_message(info, backend=send_to_admins)
  File "/home/travis/build/dimagi/commcare-hq/corehq/util/soft_assert/api.py", line 11, in _send_message
    request_repr = get_sanitized_request_repr(request)
  File "/home/travis/build/dimagi/commcare-hq/corehq/util/log.py", line 51, in get_sanitized_request_repr
    return filter.get_request_repr(request)
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/django/views/debug.py", line 120, in get_request_repr
    return build_request_repr(request, POST_override=self.get_post_parameters(request))
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/django/views/debug.py", line 184, in get_post_parameters
    return request.POST
AttributeError: 'JsonObject' object has no attribute 'POST'
```

--- a/corehq/util/log.py
+++ b/corehq/util/log.py
@@ -10,6 +10,7 @@ from pygments.formatters import HtmlFormatter
 
 from celery.utils.mail import ErrorMail
 from django.core import mail
+from django.http import HttpRequest
 from django.utils.log import AdminEmailHandler
 from django.views.debug import SafeExceptionReporterFilter, get_exception_reporter_filter
 from django.template.loader import render_to_string
@@ -44,11 +45,11 @@ def get_sanitized_request_repr(request):
     """
     Santizes sensitive data inside request object
     """
-    if not request:
-        return "Request repr() unavailable"
+    if isinstance(request, HttpRequest):
+        filter = get_exception_reporter_filter(request)
+        return filter.get_request_repr(request)
 
-    filter = get_exception_reporter_filter(request)
-    return filter.get_request_repr(request)
+    return request
 
 
 class HqAdminEmailHandler(AdminEmailHandler):
@@ -62,8 +63,6 @@ class HqAdminEmailHandler(AdminEmailHandler):
     )
     """
     def get_context(self, record):
-        request = None
-
         try:
             request = record.request
         except Exception:

--- a/corehq/util/tests/test_soft_assert.py
+++ b/corehq/util/tests/test_soft_assert.py
@@ -1,5 +1,7 @@
 import math
-from django.test import SimpleTestCase
+from django.test import SimpleTestCase, RequestFactory
+from django.test.utils import override_settings
+from corehq.util.log import get_sanitized_request_repr
 from corehq.util.soft_assert.core import SoftAssert
 from corehq.util.soft_assert.api import _send_message
 from corehq.util.cache_utils import ExponentialBackoff
@@ -63,6 +65,19 @@ class SoftAssertHelpersTest(SimpleTestCase):
             expected = n in powers_of_two
             self.assertEqual(actual, expected,
                              '_number_is_power_of_two: {}'.format(actual))
+
+    @override_settings(DEBUG=False)
+    def test_request_sanitization(self):
+        raw_request = RequestFactory().post('/accounts/login/', {'username': 'sreddy', 'password': 'mypass'})
+        raw_request.sensitive_post_parameters = '__ALL__'
+        santized_request = get_sanitized_request_repr(raw_request)
+
+        # raw request exposes password
+        self.assertTrue('mypass' in str(raw_request))
+
+        # sanitized_request should't expose password
+        self.assertFalse('mypass' in santized_request)
+        self.assertTrue('*******' in santized_request)
 
     def test_send_message(self):
 

--- a/corehq/util/tests/test_soft_assert.py
+++ b/corehq/util/tests/test_soft_assert.py
@@ -69,6 +69,7 @@ class SoftAssertHelpersTest(SimpleTestCase):
     @override_settings(DEBUG=False)
     def test_request_sanitization(self):
         raw_request = RequestFactory().post('/accounts/login/', {'username': 'sreddy', 'password': 'mypass'})
+        # Django setting to mark request sensitive
         raw_request.sensitive_post_parameters = '__ALL__'
         santized_request = get_sanitized_request_repr(raw_request)
 

